correct_periscope_drift.py

```diff
@@ -0,0 +1,431 @@
+    :returns: transform matrix
+    :rtype: Nx3x3 numpy array
+
+    """
+    ra = np.radians(ra)
+    dec = np.radians(dec)
+    roll = np.radians(roll)
+    ca = np.cos(ra)
+    sa = np.sin(ra)
+    cd = np.cos(dec)
+    sd = np.sin(dec)
+    cr = np.cos(roll)
+    sr = np.sin(roll)
+    # This is the transpose of the transformation matrix (related to
+    # translation of original perl code
+    rmat = np.array(
+        [[ca * cd,                  sa * cd,                sd     ],
+         [-ca * sd * sr - sa * cr, -sa * sd * sr + ca * cr, cd * sr],
+         [-ca * sd * cr + sa * sr, -sa * sd * cr - ca * sr, cd * cr]])
+
+    return rmat.transpose()
+
+
+def radec2eci(ra, dec):
+    """
+    Convert from RA,Dec to ECI.  The input ``ra`` and ``dec`` values can be 1-d
+    arrays of length N in which case the output ``ECI`` will be an array with
+    shape (3,N).
+
+    Borrowed from Ska.quatutil
+
+    :param ra: Right Ascension (degrees)
+    :param dec: Declination (degrees)
+    :returns: numpy array ECI (3-vector or 3xN array)
+    """
+    r = np.radians(ra)
+    d = np.radians(dec)
+    return np.array([np.cos(r) * np.cos(d), np.sin(r) * np.cos(d), np.sin(d)])
+
+
+#def roll_range(asp_roll):
+#    max_roll = np.max(asp_roll)
+#    min_roll = np.min(asp_roll)
+#    if (max_roll - min_roll) > 180:
+#        roll_180 = asp_roll.copy()
+#        roll_180[roll_180 > 180] -= 360
+#        max_roll = np.max(roll_180)
+#        min_roll = np.min(roll_180)
+#        if max_roll - min_roll > 180:
+#            raise ValueError("> 180 deg roll range")
+#    return max_roll - min_roll
+#
+
+def extract_events(event_file, src_x, src_y, src_radius):
+    """
+    Get events from specified source circle
+
+    :param event_file: Chandra event 1 or 2 file
+    :param src_x: Sky X coordinate of source region to extract
+    :param src_y: Sky Y coordinate of source region to extract
+    :param src_radius: Source region/circle radius in pixels
+    :returns: CRATE of events
+    """
+    import pycrates
+    regstring = "circle({},{},{})".format(src_x, src_y, src_radius)
+    events = pycrates.read_file("{}[sky={}]".format(
+            event_file, regstring))
+    return events
+
+
+def get_event_yag_zag(evt_ra, evt_dec, ra_nom, dec_nom, roll_nom):
+    """
+    Convert RA and Dec positions into Y and Z angle
+
+    This takes the RA and Dec positions of the events and takes a "nominal" pointing
+    (which maybe taken from the nominal values provided in the event file header
+    and uses that reference nominal pointing to convert to Y and Z angle
+    that should relate to Aspect Camera Y and Z angle for the purposes of
+    fitting Y and Z angle periscope drift.
+
+    :param evt_ra: event RA
+    :param evt_dec: event Dec
+    :param ra_nom: A single "nominal" reference value for RA
+    :param dec_nom: A single "nominal" reference value for Dec
+    :param roll_nom: A single "nominal" reference value for Roll
+    :returns: yag, zag in arcsecs
```

Comment:
Do you mean that the yag and zag should be more directly defined as "Aspect Camer Y and Z angle" in the "returns" text?  This also isn't a function the user should be directly using.